lib/libzfs/libzfs_dataset.c

```diff
@@ -2287,7 +2288,18 @@ get_numeric_property(zfs_handle_t *zhp, zfs_prop_t prop, zprop_source_t *src,
 		switch (zfs_prop_get_type(prop)) {
 		case PROP_TYPE_NUMBER:
 		case PROP_TYPE_INDEX:
-			*val = getprop_uint64(zhp, prop, source);
+			/*
+			 * We can directly read createtxg property from zfs
+			 * handle for filesystem and snapshot types.
+			 */
+			if ((prop == ZFS_PROP_CREATETXG) &&
```

Comment:
This should go in the `switch (prop)` block rather than checking `prop` again here. It can fall through with `zfs_fallthrough;` for bookmarks.

Reply:
For that matter, I wonder why there are no bookmark checks for `dds_redacted` and `dds_num_clones` and `dds_inconsistent`. Do we really need to check `zfs_type`?

Reply:
@freqlabs  zfs bookmark handle is created under ```make_bookmark_handle()```, however, ```zfs_dmustats``` is not directly assigned there. ```createtxg``` property is being used by the bookmarks, not sure about ```dds_redacted``` and ```dds_num_clones```.